File: ipywe/helloworld2.py
import ipywidgets as widgets
from traitlets import Unicode


@widgets.register('hello.Hello2')
class HelloWorld2(widgets.DOMWidget):
    """"""
    _view_name = Unicode('HelloView2').tag(sync=True)
    _model_name = Unicode('HelloModel2').tag(sync=True)
    _view_module = Unicode('ipywe').tag(sync=True)
    _model_module = Unicode('ipywe').tag(sync=True)
    _view_module_version = Unicode('^0.1.0').tag(sync=True)
    _model_module_version = Unicode('^0.1.0').tag(sync=True)
    value = Unicode('Hello World 2!').tag(sync=True)

File: js/src/helloworld2.js
var widgets = require('jupyter-js-widgets');
var _ = require('underscore');


// Custom Model. Custom widgets models must at least provide default values
// for model attributes, including
//
//  - `_view_name`
//  - `_view_module`
//  - `_view_module_version`
//
//  - `_model_name`
//  - `_model_module`
//  - `_model_module_version`
//
//  when different from the base class.

// When serialiazing the entire widget state for embedding, only values that
// differ from the defaults will be specified.
var HelloModel2 = widgets.DOMWidgetModel.extend({
    defaults: _.extend(_.result(this, 'widgets.DOMWidgetModel.prototype.defaults'), {
        _model_name : 'HelloModel2',
        _view_name : 'HelloView2',
        _model_module : 'ipywe',
        _view_module : 'ipywe',
        _model_module_version : '0.1.0',
        _view_module_version : '0.1.0',
    })
});


// Custom View. Renders the widget model.
var HelloView2 = widgets.DOMWidgetView.extend({
    render: function() {
        this.value_changed();
        this.model.on('change:value', this.value_changed, this);
    },

    value_changed: function() {
        this.el.textContent = this.model.get('value');
    }
});


module.exports = {
    HelloModel2 : HelloModel2,
    HelloView2 : HelloView2
};
